mapper.go

```diff
@@ -156,12 +158,22 @@ func (m *metricMapper) initFromYAMLString(fileContents string) error {
 		if _, ok := currentMapping.Labels["name"]; !ok {
 			return fmt.Errorf("line %d: metric mapping didn't set a metric name", i)
 		}
+		if currentMapping.MatchType == "" {
+			currentMapping.MatchType = n.Defaults.MatchType
+		}
 
-		// Translate the glob-style metric match line into a proper regex that we
-		// can use to match metrics later on.
-		metricRe := strings.Replace(currentMapping.Match, ".", "\\.", -1)
-		metricRe = strings.Replace(metricRe, "*", "([^.]+)", -1)
-		currentMapping.regex = regexp.MustCompile("^" + metricRe + "$")
+		if currentMapping.MatchType == matchTypeGlob {
```

Comment:
I'd have used a case statement, switching on the value of `currentMapping.MatchType` and an explicit default case raising an error on an unknown matchtype.